test/kube_testing/k8s.go

```diff
@@ -603,3 +613,30 @@ func (o *K8s) CleanupConfigMaps() {
 		_ = o.clientset.CoreV1().ConfigMaps("default").Delete(cm.Name, &metaV1.DeleteOptions{})
 	}
 }
+
+func (o *K8s) CreateRoles(rolesList ...string) ([]nsmrbac.Role, error) {
+	createdRoles := []nsmrbac.Role{}
+	for _, kind := range rolesList {
+		role := nsmrbac.Roles[kind](nsmrbac.RoleNames[kind])
+		err := role.Create(o.clientset)
+		if err != nil {
+			logrus.Errorf("failed creating role: %v %v", role, err)
+			return createdRoles, err
+		} else {
+			logrus.Infof("role is created: %v", role)
+			createdRoles = append(createdRoles, role)
+		}
+	}
+	return createdRoles, nil
+}
+
+func (o *K8s) DeleteRoles(rolesList []nsmrbac.Role) error {
+	for i := range rolesList {
+		err := rolesList[i].Delete(o.clientset, rolesList[i].GetName())
+		if err != nil {
```

Comment:
Well, what about the other pendint for deletion roles? You surely want to try deletion there too.